pkg/backend/httpstate/client/ai.go

```diff
@@ -54,8 +66,37 @@ func createSummarizeUpdateRequest(
 	}
 }
 
-// extractSummaryFromResponse parses the Copilot API response and extracts the summary content
-func extractSummaryFromResponse(copilotResp apitype.CopilotSummarizeUpdateResponse) (string, error) {
+// createExplainPreviewRequest creates a new CopilotExplainPreviewRequest with the given content and org ID
+func createExplainPreviewRequest(
+	lines []string,
+	orgID string,
+	maxUpdateOutputLen int,
+) apitype.CopilotExplainPreviewRequest {
+	content := strings.Join(lines, "\n")
+	content = TruncateWithMiddleOut(content, maxUpdateOutputLen)
```

Comment:
I feel like this is asking for trouble for the most interesting (because they are long and complex) previews. To make it great, I assume we need to combine the rendering logic with max limit and intelligently drop the least important events (or their details). The transport layer here can't do a good job looking at the plain string.

Would it make sense to move the message limit to the `display/ai` logic?

Reply:
This sounds very cool but I'm not sure how to filter out the less interesting events. This would in effect be trying to figure out which are less interesting _resources_ and dropping their diffs.

I suppose we could render into two different buffers. One for things we know are interesting and another for potentially less interesting things. Then add rendered events from the "less interesting" list until we hit the limit.

Again something I'd postpone for a follow up I think.